packages/php-wasm/universal/src/lib/base-php.ts

```diff
@@ -264,21 +265,28 @@ export abstract class BasePHP implements IsomorphicLocalPHP {
 				this.#setPHPCode(' ?>' + request.code);
 			}
 			this.#addServerGlobalEntriesInWasm();
-			const response = await this.#handleRequest();
-			if (request.throwOnError && response.exitCode !== 0) {
-				const output = {
-					stdout: response.text,
-					stderr: response.errors,
-				};
-				console.warn(`PHP.run() output was:`, output);
-				const error = new Error(
-					`PHP.run() failed with exit code ${response.exitCode} and the following output`
+			try {
+				const response = await this.#handleRequest();
+				if (request.throwOnError && response.exitCode !== 0) {
+					const output = {
+						stdout: response.text,
+						stderr: response.errors,
+					};
+					console.warn(`PHP.run() output was:`, output);
+					const error = new Error(
+						`PHP.run() failed with exit code ${response.exitCode} and the following output`
+					);
+					// @ts-ignore
+					error.output = output;
+					throw error;
+				}
+				return response;
+			} catch (e: any) {
+				get_logger(this).log(
```

Comment:
This will run in the browser, Node.js, VS Code, etc. I'd stick to emitting events from this method and handling environment-specific logic in an event handler. You could register one in the `website` package if the logging only matters when we can display the modal, or in the `remote` package to log data even when Playground is embedded in another app.

Reply:
Ah, I see, that was the idea so that the error information can be exposed in any environment! Hm. I'd still emit that as an event, BasePHP doesn't need to know about the logger.